server/src/main/java/org/opensearch/indices/replication/checkpoint/SegmentReplicationCheckpointPublisher.java

```diff
@@ -34,7 +34,9 @@ public SegmentReplicationCheckpointPublisher(PublishAction publishAction) {
 
     public void publish(IndexShard indexShard, ReplicationCheckpoint checkpoint) {
         publishAction.publish(indexShard, checkpoint);
-        indexShard.onCheckpointPublished(checkpoint);
+        if (indexShard.indexSettings().isSegRepWithRemoteEnabled() == false) {
```

Comment:
Would like to avoid these conditions, if you are using the publisher it should always invoke onCheckpointPublished.

Should we add a new metric here for remote store instead? I'm not sure it makes sense that the upload time is included in "replication lag."

Reply:
Either that or let's decouple this from the publisher entirely and invoke the method on shard directly in both cases.

Reply:
Got it. I've added `indexShard.onCheckpointPublished(checkpoint)` unconditionally in both the refresh listeners. 

Incase `SegmentReplicationCheckpointPublisher` and `RemoteStoreRefreshListener`, both are enabled(remote store and node-node segrep), both will invoke the same call but the method is idempotent so it should be safe.